launch/launch/launcher.py

```diff
@@ -225,7 +254,18 @@ def _run(self):
                         except ProcessLookupError:
                             pass
 
-            yield from asyncio.wait(all_futures.keys())
+            kwargs = {}
+            if not isinstance(threading.current_thread(), threading._MainThread):
+                # wake up periodically if we are not in the main thread
```

Comment:
When a launch file is being executed directly the semantic of `_MainThread` is clear and this works. But what if a Python script starts a new thread and calls `launch` in there? I would expect the check for `_MainThread` to not work in this case.